homeassistant/components/modbus/switch.py

```diff
@@ -195,49 +248,21 @@ def __init__(self, hub: ModbusHub, config: dict[str, Any]):
 
     def turn_on(self, **kwargs):
         """Set switch on."""
-        self._write_coil(self._coil, True)
+        self._write_modbus(self._coil, True)
         self._is_on = True
         self.schedule_update_ha_state()
 
     def turn_off(self, **kwargs):
         """Set switch off."""
-        self._write_coil(self._coil, False)
+        self._write_modbus(self._coil, False)
```

Comment:
This is not correct...there are not write_modbus only write_register and write_coil